### ISSUE TITLE
Docker compose example fails to start

### ISSUE THREAD

**Issue by vincentjames501** [envoyproxy/ratelimit#188]
Cloning this repository and running:

```
docker-compose -f docker-compose-example.yml
```

Is failing to start for me with the following errors:

```
envoy-proxy_1  | [2020-11-12 22:09:21.984][1][critical][main] [source/server/server.cc:104] error initializing configuration '/etc/envoy/envoy.yaml': The v2 xDS major version is deprecated and disabled by default. Support for v2 will be removed from Envoy at the start of Q1 2021. You may make use of v2 in Q3 2020 by setting '--bootstrap-version 2' on the CLI for a v2 bootstrap file and also enabling the runtime envoy.reloadable_features.enable_deprecated_v2_api flag. (Unknown field in: {"static_resources":{"listeners":[{"filter_chains":[{"filters":[{"config":{"stat_prefix":"ingress","codec_type":"auto","http_filters":[{"config":{"domain":"rl","rate_limit_service":{"grpc_service":{"envoy_grpc":{"cluster_name":"ratelimit"}}},"request_type":"external","failure_mode_deny":false,"stage":0,"rate_limited_as_resource_exhausted":true},"name":"envoy.rate_limit"},{"name":"envoy.router","config":{}}],"route_config":{"name":"route","virtual_hosts":[{"name":"backend","domains":["*"],"routes":[{"route":{"cluster":"mock","rate_limits":[{"actions":[{"source_cluster":{}},{"destination_cluster":{}}]}]},"match":{"prefix":"/test"}},{"route":{"rate_limits":[{"actions":[{"request_headers":{"header_name":"foo","descriptor_key":"foo"}}]}],"cluster":"mock"},"match":{"prefix":"/header"}},{"match":{"prefix":"/twoheader"},"route":{"cluster":"mock","rate_limits":[{"actions":[{"request_headers":{"descriptor_key":"foo","header_name":"foo"}},{"request_headers":{"descriptor_key":"bar","header_name":"bar"}}]},{"actions":[{"request_headers":{"header_name":"foo","descriptor_key":"foo"}},{"request_headers":{"header_name":"baz","descriptor_key":"baz"}}]}]}}]}]}},"name":"envoy.http_connection_manager"}]}],"address":{"socket_address":{"port_value":8888,"address":"0.0.0.0"}}}],"clusters":[{"connect_timeout":"1s","type":"STRICT_DNS","http2_protocol_options":{},"load_assignment":{"cluster_name":"ratelimit","endpoints":[{"lb_endpoints":[{"endpoint":{"address":{"socket_address":{"port_value":8081,"address":"ratelimit"}}}}]}]},"name":"ratelimit","protocol_selection":"USE_CONFIGURED_PROTOCOL","lb_policy":"ROUND_ROBIN"},{"load_assignment":{"endpoints":[{"lb_endpoints":[{"endpoint":{"address":{"socket_address":{"port_value":9999,"address":"envoy-mock"}}}}]}],"cluster_name":"mock"},"name":"mock","type":"STRICT_DNS","lb_policy":"ROUND_ROBIN","connect_timeout":"1s"}]},"admin":{"address":{"socket_address":{"port_value":8001,"address":"0.0.0.0"}},"access_log_path":"/dev/null"}})
envoy-proxy_1  | [2020-11-12 22:09:21.984][1][info][main] [source/server/server.cc:734] exiting
```

And mock:

```
envoy-mock_1   | [2020-11-12 22:09:23.007][1][critical][main] [source/server/server.cc:104] error initializing configuration '/etc/envoy/envoy.yaml': The v2 xDS major version is deprecated and disabled by default. Support for v2 will be removed from Envoy at the start of Q1 2021. You may make use of v2 in Q3 2020 by setting '--bootstrap-version 2' on the CLI for a v2 bootstrap file and also enabling the runtime envoy.reloadable_features.enable_deprecated_v2_api flag. (Unknown field in: {"admin":{"address":{"socket_address":{"address":"0.0.0.0","port_value":8001}},"access_log_path":"/dev/null"},"static_resources":{"listeners":[{"address":{"socket_address":{"address":"0.0.0.0","port_value":9999}},"filter_chains":[{"filters":[{"name":"envoy.http_connection_manager","config":{"route_config":{"virtual_hosts":[{"routes":[{"match":{"prefix":"/"},"direct_response":{"status":"200","body":{"inline_string":"Hello World"}}}],"name":"backend","domains":["*"]}],"name":"ingress"},"stat_prefix":"ingress","codec_type":"auto","http_filters":[{"name":"envoy.router","config":{}}]}}]}]}]}})
envoy-mock_1   | [2020-11-12 22:09:23.007][1][info][main] [source/server/server.cc:734] exiting
```

It looks like if I change the envoy-dev version to something slightly older like `baa1417b24c2d10b17c4c05c2211f6cfb4ba3b82` it appears to work fine again.

I'm trying to use the new `typed_config` key as well but not having much luck. Any guidance there?

**Comment by ysawa0** [comment]
I have a PR open on my fork to update the config to V3. https://github.com/ysawa0/ratelimit/pull/5
@vincentjames501 Can you try the configs in there and see if it works?

I hadn't gotten around to merging it into to upstream but let me open that up once you confirm it's good.

**Comment by vincentjames501** [comment]
@ysawa0 , I'm seeing the following on startup on ysawa0#5

```
envoy-proxy_1  | Protobuf message (type envoy.config.bootstrap.v3.Bootstrap reason INVALID_ARGUMENT:(http_filters[0]) config: Cannot find field.) has unknown fields
ratelimit_envoy-proxy_1 exited with code 1
```

It seems like we need to specify a `@type` and `type_config` for the `envoy.rate_limit` filter? I just couldn't figure out what to set it to.

> Disclaimer, I just started tinkering with Envoy for the first time today.

**Comment by ysawa0** [comment]
@vincentjames501 Thanks, I pushed a fix for that and opened the PR #189

**Comment by vincentjames501** [comment]
Thanks @ysawa0 , appears to work!
